### ISSUE TITLE
Arch linux will not fetch

### ISSUE THREAD

**Issue by fosslinux** [bedrocklinux/bedrocklinux-userland#155]
```
➜ sudo brl fetch arch
[ 1/19 (  5%)] Determining name
* Using arch
[ 2/19 ( 10%)] Determining CPU architecture
* Using x86_64
[ 3/19 ( 15%)] Determining release
* Using rolling
[ 4/19 ( 21%)] Determining mirror
* Checking bedrock.conf [brl-fetch-mirror] items
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
* Fast filtering 169 to top 10
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
* Finding fastest mirror from 10
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
* Using https://mirror.osbeck.com/archlinux/
[ 5/19 ( 26%)] Making bootstrap directory structure
[ 6/19 ( 31%)] Downloading package information database
Connecting to mirror.osbeck.com (104.25.194.31:443)
saving to '/bedrock/strata/arch/brl-bootstrap/core.db.tar.gz'
core.db.tar.gz       100% |********************************************************************************************************************************************************************|  134k  0:00:00 ETA
'/bedrock/strata/arch/brl-bootstrap/core.db.tar.gz' saved
Connecting to mirror.osbeck.com (104.25.194.31:443)
saving to '/bedrock/strata/arch/brl-bootstrap/extra.db.tar.gz'
extra.db.tar.gz      100% |********************************************************************************************************************************************************************| 1639k  0:00:00 ETA
'/bedrock/strata/arch/brl-bootstrap/extra.db.tar.gz' saved
[ 7/19 ( 36%)] Decompressing package information database
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
[ 8/19 ( 42%)] Converting distro package information database to brl format
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
[ 9/19 ( 47%)] Calculating required bootstrap packages
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
[10/19 ( 52%)] Downloading bootstrap packages
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
[11/19 ( 57%)] Extracting bootstrap packages
[\\\\\                           ]  13%tar: short read
[\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\] 100%
ERROR: Unexpected error occurred.
```

I am at a loss to why this is occurring. I have tried various mirrors and none of them are working. `tar` is from void if it is of any significance.

**Comment by nilesr** [comment]
I'm not in a position to give good directions right now but it would be
great to get an strace of the file fetch, a stat of the file in question
before tar is invoked, and an strace of the tar. The script is in
/bedrock/libexec/brl-fetch/arch IIRC

**Comment by paradigm** [comment]
https://www.archlinux.org/news/required-update-to-recent-libarchive/

Arch recently updated its package format compression from `xz` to `zstd`.  I missed this and didn't update `brl fetch` accordingly.  What's happening here is it's decompressing with `xz` then piping the broken results to `tar` which is understandably confused.

Fedora 31 did the same thing a number of months back, and I added `zstd` to Bedrock then.  All that's needed here and now is to make `extract_pacman_pkgs()` in `/bedrock/libexec/brl-fetch` check the file extension - `.xz` or `.zst` - decompress the package appropriately.  Should be a quick easy fix.

I also have a fix for `brl fetch kiss` queued up as well.  I'll see if I can bundle those two into 0.7.13 this weekend.

-----

> tar is from void if it is of any significance.

It's probably actually from `/bedrock/libexec/busybox`, actually.  Bedrock provides its own copy of things needed for `brl fetch` to avoid dependency cycles where you need to `brl fetch` something to get a dependency for `brl fetch`.

**Comment by paradigm** [comment]
Anyone who needs to `brl fetch arch` before the update is out, open up `/bedrock/libexec/brl-fetch` and change [this line](https://github.com/bedrocklinux/bedrocklinux-userland/blob/master/src/slash-bedrock/libexec/brl-fetch#L602):

```
		unxz <"${file}" | tar -xf - -C "${dir}"
```

to

```
		if echo "${file}" | grep "[.]xz"; then
			unxz <"${file}" | tar -xf - -C "${dir}"
		elif echo "${file}" | grep "[.]zst"; then
			/bedrock/libexec/zstd -d <"${file}" | tar -xf - -C "${dir}"
		else
			abort \"Unrecognized file extension: ${file}\"
		fi
```

**Comment by fosslinux** [comment]
Cool, thanks @paradigm !

**Comment by paradigm** [comment]
Fixed with 0.7.13
